fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanRepaymentBusinessEventSerializer.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.event.external.service.serialization.serializer.loan;
+
+import java.math.BigDecimal;
+import lombok.RequiredArgsConstructor;
+import org.apache.avro.generic.GenericContainer;
+import org.apache.fineract.avro.generator.ByteBufferSerializable;
+import org.apache.fineract.avro.generic.v1.CurrencyDataV1;
+import org.apache.fineract.avro.loan.v1.LoanRepaymentDueDataV1;
+import org.apache.fineract.avro.loan.v1.RepaymentDueDataV1;
+import org.apache.fineract.infrastructure.event.business.domain.BusinessEvent;
+import org.apache.fineract.infrastructure.event.business.domain.loan.repayment.LoanRepaymentBusinessEvent;
+import org.apache.fineract.infrastructure.event.external.service.serialization.serializer.AbstractBusinessEventSerializer;
+import org.apache.fineract.organisation.monetary.domain.MonetaryCurrency;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanRepaymentScheduleInstallment;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class LoanRepaymentBusinessEventSerializer extends AbstractBusinessEventSerializer {
+
+    @Override
+    protected <T> ByteBufferSerializable toAvroDTO(BusinessEvent<T> rawEvent) {
+        LoanRepaymentBusinessEvent event = (LoanRepaymentBusinessEvent) rawEvent;
+        LoanRepaymentScheduleInstallment repaymentInstallment = event.get();
+        Loan loan = repaymentInstallment.getLoan();
+        Long id = loan.getId();
+        String accountNo = loan.getAccountNumber();
+        String externalId = loan.getExternalId();
+        String dueDate = repaymentInstallment.getDueDate().toString();
```

Comment:
ToString is wrong here. The format for dates and timestamps are specified in Avro. The MapStruct mappers are using AvroDateTimeMapper to map dates and times and you can do the same thing here.